datastream-kafka/src/main/java/com/linkedin/datastream/kafka/KafkaTransportProvider.java

```diff
@@ -161,8 +173,13 @@ public void send(String destinationUri, DatastreamProducerRecord record, SendCal
         producer.send(_datastreamTask, outgoing, (metadata, exception) -> {
           int partition = metadata != null ? metadata.partition() : -1;
           if (exception != null) {
-            LOG.error("Sending a message with source checkpoint {} to topic {} partition {} for datastream task {} "
-                    + "threw an exception.", record.getCheckpoint(), topicName, partition, _datastreamTask, exception);
+            if (_isUnassigned) {
+              LOG.debug("Sending a message with source checkpoint {} to topic {} partition {} for datastream task {} "
```

Comment:
The log string is the same, can you construct the string outside the if-else, and just use that variable to print?